specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/deviceSecurityGroups.json

```diff
@@ -0,0 +1,559 @@
+            "values": [
+              {
+                "value": "IpCidr",
+                "description": "An IP range in CIDR format (e.g. '192.168.0.1/8')."
+              },
+              {
+                "value": "String",
+                "description": "Any string value."
+              }
+            ]
+          }
+        }
+      }
+    },
+    "AllowlistCustomAlertRule": {
+      "type": "object",
+      "description": "A custom alert rule that checks if a value (depends on the custom alert type) is allowed.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ListCustomAlertRule"
+        }
+      ],
+      "properties": {
+        "allowlistValues": {
+          "type": "array",
+          "description": "The values to allow. The format of the values depends on the rule type.",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ruleType": {
+          "type": "string",
+          "description": "The type of the custom alert rule.",
+          "enum": [
+            "ConnectionToIpNotAllowed",
+            "LocalUserNotAllowed",
+            "ProcessNotAllowed"
+          ],
+          "x-ms-enum": {
+            "name": "AllowListRuleType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "ConnectionToIpNotAllowed",
+                "description": "Outbound connection to an ip that isn't allowed. Allow list consists of ipv4 or ipv6 range in CIDR notation."
+              },
+              {
+                "value": "LocalUserNotAllowed",
+                "description": "Login by a local user that isn't allowed. Allow list consists of login names to allow."
+              },
+              {
+                "value": "ProcessNotAllowed",
+                "description": "Execution of a process that isn't allowed. Allow list consists of process names to allow."
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "allowlistValues"
+      ]
+    },
+    "DenylistCustomAlertRule": {
+      "type": "object",
+      "description": "A custom alert rule that checks if a value (depends on the custom alert type) is denied.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ListCustomAlertRule"
+        }
+      ],
+      "properties": {
+        "denylistValues": {
+          "type": "array",
+          "description": "The values to deny. The format of the values depends on the rule type.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "denylistValues"
+      ]
+    },
+    "ThresholdCustomAlertRule": {
+      "type": "object",
+      "description": "A custom alert rule that checks if a value (depends on the custom alert type) is within the given range.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CustomAlertRule"
+        }
+      ],
+      "properties": {
+        "minThreshold": {
```

Comment:
Any min max value for the threshold that can be client validated?

Reply:
There is no limitation for the customer number